### PR TITLE
Fix story skip

### LINKED ISSUE #830: Skip Story doesn't work
### Preparation

- [X] I tested the [latest release](https://github.com/Fate-Grand-Automata/FGA/releases/tag/canary)
- [X] I looked at other issues (even the [closed ones](https://github.com/Fate-Grand-Automata/FGA/issues?q=is%3Aissue+is%3Aclosed))
- [X] I read the [Troubleshooting Guide](https://github.com/MathewSachin/Fate-Grand-Automata/wiki/Troubleshooting)

### FGO server

NA

### FGA build number

1468

### Describe the bug

Playing on Bluestacks 5. In the Dubug Mode I can see that the program does detect the story skip button but just doesn't press it. I tried restarting, reinstalling, regiving the permissions.  

### Screenshots

_No response_

### Video

_No response_

### Device model

OnePlus 3T

### Android version

Android 9

### Screen size

_No response_

### RAM

_No response_

### PR DESCRIPTION
**Pull request by reconman**
Fixes #830 

This sets the background of the story skip button to black.

The code change is related to a discovery that some images with half-transparent pixels. OpenCV runs a `threshold` with `0` as the threshold, so barely visible pixels are treated as fully visible.

Even after the code change, story skip's score is always a bunch of `-Infinity` on a black screen with a Skip button. Changing the background to black at least finds the image in some quests.

### CODE REVIEW

**Comment by MathewSachin** [comment]
> OpenCV runs a threshold with 0 as the threshold, so barely visible pixels are treated as fully visible.

Can you point me to their code base where they do this. Sounds really bad.

**Comment by reconman** [comment]
https://github.com/opencv/opencv/blob/master/modules/imgproc/src/templmatch.cpp#L783

**Comment by MathewSachin** [comment]
So, 8U masks are binary.
How about we use 32F then?

**Comment by MathewSachin** [comment]
Just do a `convertTo` like they're doing after thresholding and we won't need to binary threshold.

**Comment by reconman** [comment]
Can you find the implementation of convertTo?

**Comment by reconman** [comment]
We can try `alphaChannel.convertTo(alphaChannel, CV_32F, 1.0/255)` from https://docs.opencv.org/4.5.2/d6/d39/samples_2dnn_2colorization_8cpp-example.html#a13

I don't have great hopes, though.

**Comment by reconman** [comment]
Didn't help, I opened a bug report on the OpenCV project: https://github.com/opencv/opencv/issues/20557

**Comment by reconman** [comment]
In the python script for reproducing the issue, I've changed the matching algorithm to CCORR_NORMED for testing and got good match results.

I tested matching with story skip and skill 10. Skill 10 matched some level 1 skills so I increased the min similarity from 0.8 to 0.9. I was getting values around 0.85 for "1".

I could also remove the "1" again from the skill-ten image, then the similarity would go down for skills with level "1".

**Comment by KnightRevan** [comment]
Is this solution applicable to the apk version and if so, how do I make this change?

File: app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt [hunk 1]
```diff
@@ -208,7 +212,7 @@ class DroidCvPattern(
     override fun threshold(value: Double): IPattern {
         val result = Mat()
 
-        Imgproc.threshold(mat, result, value * 255, 255.0, Imgproc.THRESH_BINARY)
+        Imgproc.threshold(mat, result, value * 255, 255.0, THRESH_BINARY)
```
Comment (MathewSachin) [inline app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt:215]: nitpick: revert back to `Imgproc.THRESH_BINARY` and remove the import. only used at 1 place now.

Comment (reconman) [reply app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt]: Done

**Review by MathewSachin — APPROVED** [review]

**Pull request merged**